src/engine/ItemmarkManager.ts

```diff
@@ -0,0 +1,105 @@
+import type { IMediaContainer } from './providers/MediaPlugin';
+import { type StorageController, Store } from './StorageController';
+
+export class ItemmarkManager {
+
+    private readonly Items:Map<string,ItemMark[]>=new Map();
+    constructor(private readonly storage: StorageController) {
+        this.Load();
+    }
+
+    private async Load() {
+        const mediamarks = await this.storage.LoadPersistent<ItemmarkSerialized[]>(Store.Itemmarks);
+        mediamarks.forEach(itemmark => {
+            this.Items.set(`${itemmark.Media.ProviderID} :: ${itemmark.Media.EntryID}`,itemmark.Marks);
+        });
+    }
+
+    private Serialize(media: IMediaContainer, marks: ItemMark[]): ItemmarkSerialized {
+        return {
+            Media: {
+                ProviderID: media.Parent.Parent.Identifier,
+                EntryID: media.Parent.Identifier
+            },
+            Marks: marks,
+        };
+    }
+
+    /* Retrieves the marks of a media */
+    public Find(entry: IMediaContainer): ItemMark[] {
+        return this.Items.get(this.StorageKey(entry));
+    }
+
+    /* Add a mark of an item */
+    public async Add(entry: IMediaContainer, kind: MarkType): Promise<ItemMark[]> {
+        const newmark: ItemMark = {
```

Comment:
Should not be allowed to add `MarkType.None` => save storgae space

Reply:
None isn't stored, it's returned by getMark if none ... but i could have returned null instead ... #Brain